File: Practicas/Marco-Oregel/Practica7/MarvelCharacter.py
from typing import Protocol

class MarvelCharacter(Protocol):

    def __init__(self,name,alias,universe,age):
        self.name = name
        self.alias = alias
        self.universe = universe
        self._age = age
        

    def showPower(self):
        print(f"Pow,Boom,Kapow!")


    def getAge(self):
        return f"{self.name} is {self._age} years old"
    
    def updateAge(self, newAge):
        if self._age > 0:
            self._age = newAge
            print("Age has been updated!")
        else:
            print("Age must be greater than 0")

File: Practicas/Yahir-Hermosillo/Parcial3/Practica7/MarvelCharacter.java
public class MarvelCharacter {
    private String name;
    private String alias;
    private String universe;
    private int age;

    public MarvelCharacter(String name, String alias, String universe, int age) {
        this.name = name;
        this.alias = alias;
        this.universe = universe;
        if (age > 0) {
            this.age = age;
        } else {
            this.age = 0;
        }
    }

    public void showPower() {
        System.out.println("Este personaje tiene poderes especiales.");
    }

    public int getAge() {
        return age;
    }

    public void updateAge(int newAge) {
        if (newAge > 0) {
            this.age = newAge;
        }
    }
}
